code/modules/mob/living/basic/minebots/minebot_upgrades.dm

```diff
@@ -0,0 +1,60 @@
+/obj/item/mine_bot_upgrade
+	name = "minebot melee upgrade"
+	desc = "A minebot upgrade."
+	icon_state = "door_electronics"
+	icon = 'icons/obj/assemblies/module.dmi'
+
+/obj/item/mine_bot_upgrade/afterattack(mob/living/basic/mining_drone/minebot, mob/user, proximity)
+	. = ..()
+	if(!istype(minebot) || !proximity)
+		return
+	upgrade_bot(minebot, user)
+
+/obj/item/mine_bot_upgrade/proc/upgrade_bot(mob/living/basic/mining_drone/minebot, mob/user)
+	if(minebot.melee_damage_upper != initial(minebot.melee_damage_upper))
+		to_chat(user, span_warning("[minebot] already has a combat upgrade installed!"))
+		return
+	minebot.melee_damage_lower += 7
+	minebot.melee_damage_upper += 7
+	to_chat(user, "<span class='notice'>You increase the close-quarter combat abilities of [minebot].")
```

Comment:
wrap this with `span_notice()` macro instead. doesn't even close span this is so fucked up